src/Types/BooleanType.php

```diff
@@ -49,4 +50,12 @@ public function getBindingType()
     {
         return ParameterType::BOOLEAN;
     }
+
+    /**
+     * @return bool
+     */
+    public function requiresSQLCommentHint(AbstractPlatform $platform)
+    {
+        return $platform instanceof DB2Platform;
```

Comment:
https://github.com/doctrine/dbal/blob/71b2e68a35fec03326b12fc7aaf021c8c28a8f80/src/Platforms/DB2Platform.php#L99-L100

AFAIK, only PostgreSQL natively supports boolean columns but they work on all supported platforms. Is it really necessary to use a comment hint on DB2?

Reply:
I forgot to copy the comment I found in `DB2Platform`, it explains why this is necessary.

Reply:
> Is it really necessary to use a comment hint on DB2?

I have added the comment, and now I am wondering if we shouldn't require a comment on all platforms but PG

Reply:
> now I am wondering if we shouldn't require a comment on all platforms but PG

I guess it's the opposite and we don't need it anywhere (at least with the platform-aware schema comparison in place). We need to find a relevant schema comparison test and see how a scenario of deployment and comparison of a boolean column works say on MySQL.

Reply:
You read correctly, my sentence has the same meaning with or without `n't` (in my intention, now you make me realize what I wrote is properly incorrect english).

I don't understand why you say it would work. If MySQL does not have native support for boolean columns, why would the DBAL reverse engineer to `BooleanType` and not `SmallIntType`?

Reply:
I'm not certain it does work, I'm just saying I'm not aware of any related issues, so it may work somehow. We need more testing to understand the details. If it doesn't work anywhere except for PostgreSQL, then it's a separate bug.

On the other hand, the changes here are in line with the existing logic, so this unknown probably shouldn't block the deprecation.

Reply:
> then it's a separate bug.

If it's a bug, then it must be separate because it should target 3.2.x, one more reason not to block this. Plus I have other rabbit holes to dig into :P